mock/mock.go

```diff
@@ -296,7 +296,15 @@ func (m *Mock) Called(arguments ...interface{}) Arguments {
 		closestFound, closestCall := m.findClosestCall(functionName, arguments...)
 
 		if closestFound {
-			panic(fmt.Sprintf("\n\nmock: Unexpected Method Call\n-----------------------------\n\n%s\n\nThe closest call I have is: \n\n%s\n", callString(functionName, arguments, true), callString(functionName, closestCall.Arguments, true)))
+			if functionName == closestCall.Method && closestCall.Repeatability == -1 {
```

Comment:
Also FYI, `functionName == closestCall.Method` was redundant since that's already checked in `findClosestCall`.